### DISCUSSION TITLE
Download Previous Versions

### DISCUSSION THREAD

**Discussion by lllyasviel** [lllyasviel/stable-diffusion-webui-forge#849] [Announcements]
This list will be extended in real time with updates.

---

**Forge F0.0.17:**

(last tested 2024 July 21, tested all diffusion models, all reference-only methods, all controlnets, all ip-adapters, instant-id, etc.)

(Some people call this version "the old forge")

https://github.com/lllyasviel/stable-diffusion-webui-forge/releases/download/previous/webui_forge_cu121_torch21_f0017.7z

**Comment by ICodeDiffusion** [comment]
yay! :D Love your work and thank you for making forge, I wish you a great life! :D

**Comment by zhaoqi571436204** [comment]
Thank you Buddha for your hard work and look forward to a new forge

**Comment by dermesut** [comment]
two things about this "the old forge" version:
1. just tried to download it (with chrome browser).
after completion i couldn't access it, chrome-downloader says: "virus found".

2. say, above mentioned problem is gone, how is it supposed to work in terms of the update.bat?
prior to your announcement, one could download the thing and eventually click "update.bat" to get everything up to the most recent version.
now, if "update.bat" is clicked with this "old forge" version, will it update such that it becomes the "main" version (the one, that will change according to your announcement) or will it keep me "within" the "old forge" version?

**Reply by SysVR** [reply]
1.False positives.
2.main branch latest version.

**Comment by ICodeDiffusion** [comment]
> This list will be extended in real time with updates.
> 
> **Forge F0.0.17:**
> 
> (last tested 2024 July 21, tested all diffusion models, all reference-only methods, all controlnets, all ip-adapters, instant-id, etc.)
> 
> (Some people call this version "the old forge")
> 
> https://github.com/lllyasviel/stable-diffusion-webui-forge/releases/download/previous/webui_forge_cu121_torch21_f0017.7z

Hello this is for people who install the older version:

Pydantic(or something like that) will have an error when you initially run the bat file,, this fixes the issue.  You see the comment by aearone.

IGNORE the first part above the image, but head to the root folder of your webui, not where the update bat file is.


Follow the steps in CMD not terminal to activate the venv.
**I have fixed this many times just by following this:**

Source: https://github.com/Mikubill/sd-webui-controlnet/issues/2698#issuecomment-2027350047

works every time! :D

**Comment by vonespy** [comment]
Z, you're a bio-machine that cranks out the coolest toys! Thank you for all of the amazing projects you've brought into existence.

**Comment by mimimi999** [comment]


**Reply by Kuroseji** [reply]
> > This list will be extended in real time with updates.
> > **Forge F0.0.17:**
> > (last tested 2024 July 21, tested all diffusion models, all reference-only methods, all controlnets, all ip-adapters, instant-id, etc.)
> > (Some people call this version "the old forge")
> > https://github.com/lllyasviel/stable-diffusion-webui-forge/releases/download/previous/webui_forge_cu121_torch21_f0017.7z
> 
> The file is so highly compressed that there is no sign that it will finish decompressing. Could you also upload a less compressed version of the package file?

Not sure if it will help you but I was able to decompress it in about 15 minutes using WinRar instead of 7Zip. Estimated 9 hours for 7zip when I gave up on it.

**Reply by SillySilk** [reply]
same here

**Comment by SillySilk** [comment]
Getting excited for the FLUX support I see happening With the current updates. Unfortunately it broke a couple of my favorite extensions and I'm hoping as it all comes together they work again. I'm not sure if there's a place to put in a support ticket or if that is something that you even do but either way thank you so much for everything that you have done and that everyone has done to make this such an amazing ui

**Reply by Strapples** [reply]
Does queue scheduler work with the new one?

**Comment by jd-paucar** [comment]
Hi everyone!

If you want to install the old forge and the Pydantic error message appears on your CMD terminal, this is what you have to do...

- Click on the "update.bat" file. You'll upgrade your Forge to the latest version, then go to the root folder (webui) and search for the "venv" folder.
- Copy the venv folder in another location
- Delete the "webui_forge_cu121_torch21_f0017" folder to delete forge, then install it again.
- After you installed it again, paste the "venv" folder to the webui folder.
- Now, click in the "run.bat" file and it should work properly

If you have some errors when launching, go to the "webui-user.bat" file and open it with Notepads or VS code, then copy this with your python file path

![image](https://github.com/user-attachments/assets/804e7d55-a258-4052-afa6-fde60a142c2e)

**Comment by Strapples** [comment]
As far as stability goes with plugins. Let’s be honest. Is 0017 the most current “LTS” build or is there a newer one. I think the next “LTS” eligible build will hit in September or October as he did say he’s like 95% done with rewriting the backend. Sent from my iPhoneOn Aug 21, 2024, at 8:29 AM, Junior Paucar ***@***.***> wrote:﻿
Hi everyone!
If you want to install the old forge and the Pydantic error message appears on your CMD terminal, this is what you have to to...

Click on the "update.bat" file. You'll upgrade your Forge to the latest version, then go to the root folder (webui) and search for the "venv" folder.
Copy the venv folder in another location
Delete the "webui_forge_cu121_torch21_f0017" folder to delete forge, then install it again.
After you installed it again, paste the "venv" folder to the webui folder.
Now, click in the "run.bat" file and it should work properly

If you have some errors when launching, go to the "webui-user.bat" file and open it with Notepads or VS code, then copy this with your python file path
image.png (view on web)

**Comment by FCB-Sunny** [comment]
Thanks for your all efforts and can I use the regional prompter extension with flux and flux loras in this ui?

**Comment by BecomingJerome** [comment]
Hello, I really appreciate this initiative.
I wish I could find Safetensors (Checkpoints) and Loras that are compatible with "old forge". My pc specs can't run Flux and I dont want to run my drive at 100%.. I wish find SD1 models.
I'm totally lost in civitai.

**Reply by vonespy** [reply]
In Civitai, click on models, then click on filters (upper right corner), select SD 1.5 and then search. you'll get just the old SD 1.5 models. They are still there, just pushed out of the way by work on the newer models like Flux.

**Reply by BecomingJerome** [reply]
Thank you so much. vonespy.

**Reply by vonespy** [reply]
Not sure if you're wanting to run older versions of forge, but from what I can tell, the current version seems very compatible with all models, old and new. there are even radio buttons now to allow you to select if you're using sd, sdxl or flux models, which adds or takes away controls from the UI that aren't needed for those particular models.

I'm far from being an expert but feel free to send me screenshots with questions if I can help.

**Reply by BecomingJerome** [reply]
I don't remember the version I used, but I stupidly remove the archive. My PC is 6 years old with 16go of ram and 8go of vram.
It was distributed with one checkpoint with "real" in its name. I used to upscale with gigapixel and did some digital painting on it.

**Reply by vonespy** [reply]
Hopefully you have an NVIDIA GPU. You can kind of run this stuff just on CPU, but instead of a picture every few minutes, you would be getting a picture in an hour or more. 

In any case, I'd suggest you download the premade package (https://github.com/lllyasviel/stable-diffusion-webui-forge/releases/download/latest/webui_forge_cu121_torch231.7z) and install it. you'll need to also download a sd1.5 model (the safetensors files you mentioned) from civitai or other sources. You can also just download the original stock SD1.5 model and run from there. 

The GUI still isn't great but I think it's a LOT more intuitive than earlier versions. Once you get forge figured out, all of the cool kids use ComfyUI, but I haven't found that I need quite that much complexity yet.

**Reply by BecomingJerome** [reply]
Thank you again.

**Comment by 9wow** [comment]
How do you run this on macOS? The normal way of running **webui.sh** doesnt work.

**Reply by 9wow** [reply]
Nvm i got it.

cd webui folder

python3.10 -m venv venv
source venv/bin/activate

pip install -r requirements.txt

**chmod +x webui.sh**

./webui.sh

**Comment by JimmyBaevsky** [comment]
Hello, friend!

I tried all packages you provided, but got the same result: runtime error with such not very detailed diagnostic:

"Python 3.10.0 (tags/v3.10.0:b494f59, Oct  4 2021, 19:00:18) [MSC v.1929 64 bit (AMD64)]
Version: f2.0.1v1.10.1-previous-637-gdaee4c0d
Commit hash: daee4c0d8f77cd01c427b4202dd333d7f159fa19
Traceback (most recent call last):
  File "C:\Users\Jimmy\AI\Forge\webui\launch.py", line 54, in <module>
    main()
  File "C:\Users\Jimmy\AI\Forge\webui\launch.py", line 42, in main
    prepare_environment()
  File "C:\Users\Jimmy\AI\Forge\webui\modules\launch_utils.py", line 434, in prepare_environment
    raise RuntimeError(
RuntimeError: Your device does not support the current version of Torch/CUDA! Consider download another version:
https://github.com/lllyasviel/stable-diffusion-webui-forge/releases/tag/latest"

I got he same when ran package with renamed directory "system" and use Python, Torch and Cuda installed on PC.
Could you please give me some idea which steps should I do to understand the problem. Thank you in advance. Regards, Jimmy Baevsky

**Comment by nradhesh** [comment]
i wanted to create an android app using this application
can u please assist me with the backend rendering so that it can allow for request from my mobile app's frontend

**Comment by katerinna2125** [comment]
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: f2.0.1v1.10.1-previous-669-gdfdcbab6
Commit hash: dfdcbab685e57677014f05a3309b48cc87383167
Traceback (most recent call last):
  File "D:\ForgeUI\webui\launch.py", line 54, in <module>
    main()
  File "D:\ForgeUI\webui\launch.py", line 42, in main
    prepare_environment()
  File "D:\ForgeUI\webui\modules\launch_utils.py", line 434, in prepare_environment
    raise RuntimeError(
RuntimeError: Your device does not support the current version of Torch/CUDA! Consider download another version:
https://github.com/lllyasviel/stable-diffusion-webui-forge/releases/tag/latest
Для продолжения нажмите любую клавишу . . .

при запуске пишнт это, помогите пожалуйста
